File: backend/app/models/vehicle.py
from datetime import datetime

from sqlalchemy import DateTime, Enum, ForeignKey, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base
from app.models.enums import VehicleType


class Vehicle(Base):
    __tablename__ = "vehicles"

    id: Mapped[int] = mapped_column(primary_key=True)
    driver_id: Mapped[int] = mapped_column(ForeignKey("drivers.id", ondelete="CASCADE"), nullable=False)

    vehicle_type: Mapped[VehicleType] = mapped_column(Enum(VehicleType, name="vehicle_type"), nullable=False)
    registration_number: Mapped[str] = mapped_column(String(50), nullable=False)
    make: Mapped[str] = mapped_column(String(100), nullable=False)
    model: Mapped[str] = mapped_column(String(100), nullable=False)
    year: Mapped[int] = mapped_column(Integer, nullable=False)
    colour: Mapped[str] = mapped_column(String(50), nullable=False)

    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())

    driver: Mapped["Driver"] = relationship(back_populates="vehicles")

File: frontend/src/types/vehicle.ts
export type VehicleType = "MOTORCYCLE" | "BICYCLE" | "CAR" | "VAN" | "TRUCK";

export interface Vehicle {
  id: number;
  driver_id: number;
  vehicle_type: VehicleType;
  registration_number: string;
  make: string;
  model: string;
  year: number;
  colour: string;
  created_at: string;
}

export type VehicleCreate = Omit<Vehicle, "id" | "driver_id" | "created_at">;
